### PR TITLE
Added algorithm for finding index of rightmost set bit

### PR DESCRIPTION
**Pull request by GautamChaurasia**
### Describe your change:
Added algorithm for finding the zero-based index of first set bit in a given non-negative integer from the right side, i.e. rightmost set bit

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: bit_manipulation/index_of_rightmost_set_bit.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            index += 1
+        return index - 1
+    else:
+        raise ValueError("Input must be a non-negative integer")
+
+
+"""
+Finding the index of rightmost set bit has some very peculiar use-cases,
+especially in finding missing or/and repeating numbers in a list of positive integers.
+"""
```
Comment (CaedenPH) [inline bit_manipulation/index_of_rightmost_set_bit.py:39]: Move this to a docstring at the start

File: bit_manipulation/index_of_rightmost_set_bit.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+"Reference: https://www.geeksforgeeks.org/position-of-rightmost-set-bit/"
+
+
+def get_index_of_rightmost_set_bit(number: int) -> int:
+
```
Comment (CaedenPH) [inline bit_manipulation/index_of_rightmost_set_bit.py:5]:
```suggestion
```

File: bit_manipulation/index_of_rightmost_set_bit.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+"Reference: https://www.geeksforgeeks.org/position-of-rightmost-set-bit/"
```
Comment (CaedenPH) [inline bit_manipulation/index_of_rightmost_set_bit.py:1]: Make this a docstring or a comment, not a string

File: bit_manipulation/index_of_rightmost_set_bit.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    if number >= 0 and type(number) == int:
+        intermediate = number & ~(number - 1)
+        index = 0
+        while intermediate:
+            intermediate >>= 1
+            index += 1
+        return index - 1
+    else:
+        raise ValueError("Input must be a non-negative integer")
```
Comment (CaedenPH) [inline bit_manipulation/index_of_rightmost_set_bit.py:33]: you can move this to the top;

```py
if 0 > number and not isinstance(number, int):
      raise ValueError
```
And you then don't need an `else`

**Comment by GautamChaurasia** [comment]
Please review my work.

**Review by cclauss — APPROVED** [review]

**Pull request merged**